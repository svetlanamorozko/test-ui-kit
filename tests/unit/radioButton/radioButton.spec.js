import { shallow } from 'enzyme';
import React from 'react';
import RadioButton from '../../../components/radioButton/radioButton';

describe('Radio Button', () => {
  describe('#render()', () => {
    const onChange = () => {};

    it('should not modify mods', () => {
      const mods = ['test'];
      const wrapper = shallow(
        <RadioButton disabled id="disabledRadio" mods={mods} onChange={onChange}>
          Disabled radio button
        </RadioButton>
      );

      expect(mods.length).toEqual(1);
      expect(mods[0]).toEqual('test');
      expect(wrapper).toMatchSnapshot();
    });

    it('should render disabled radio button', () => {
      const wrapper = shallow(
        <RadioButton disabled id="disabledRadio" onChange={onChange}>
          Disabled radio button
        </RadioButton>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render checked radio button', () => {
      const wrapper = shallow(
        <RadioButton checked id="radio1" onChange={onChange}>
          Radio 1
        </RadioButton>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render radio button', () => {
      const wrapper = shallow(
        <RadioButton id="radio2" onChange={onChange}>
          Radio 2
        </RadioButton>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render radio button with provided className', () => {
      const wrapper = shallow(
        <RadioButton className="test-class" id="radio1" onChange={onChange}>
          Radio 1
        </RadioButton>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render radio button with dataAttrs for root element', () => {
      const wrapper = shallow(
        <RadioButton className="test-class" dataAttrs={{ 'my-attr': 'some-attr' }} id="radio1" onChange={onChange}>
          Radio 1
        </RadioButton>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render radio button with dataAttrs for input', () => {
      const wrapper = shallow(
        <RadioButton
          className="test-class"
          id="radio1"
          inputDataAttrs={{ 'my-attr-for-input': 'some-attr-for-input' }}
          onChange={onChange}
        >
          Radio 1
        </RadioButton>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render radio button with dataAttrs for label', () => {
      const wrapper = shallow(
        <RadioButton
          className="test-class"
          id="radio1"
          labelDataAttrs={{ 'my-attr-for-label': 'some-attr-for-label' }}
          onChange={onChange}
        >
          Radio 1
        </RadioButton>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
