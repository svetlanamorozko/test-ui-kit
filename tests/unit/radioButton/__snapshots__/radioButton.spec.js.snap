// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button #render() should not modify mods 1`] = `
<div
  className="ui-radio ui-radio_test ui-radio_disabled"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    disabled={true}
    id="disabledRadio"
    onChange={[Function]}
    readOnly={false}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    htmlFor="disabledRadio"
    role="radio"
  >
    Disabled radio button
  </label>
</div>
`;

exports[`Radio Button #render() should render checked radio button 1`] = `
<div
  className="ui-radio"
>
  <input
    checked={true}
    className="ui-radio__input-radio"
    disabled={false}
    id="radio1"
    onChange={[Function]}
    readOnly={false}
    type="radio"
  />
  <label
    aria-checked={true}
    className="ui-radio__label"
    htmlFor="radio1"
    role="radio"
  >
    Radio 1
  </label>
</div>
`;

exports[`Radio Button #render() should render disabled radio button 1`] = `
<div
  className="ui-radio ui-radio_disabled"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    disabled={true}
    id="disabledRadio"
    onChange={[Function]}
    readOnly={false}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    htmlFor="disabledRadio"
    role="radio"
  >
    Disabled radio button
  </label>
</div>
`;

exports[`Radio Button #render() should render radio button 1`] = `
<div
  className="ui-radio"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    disabled={false}
    id="radio2"
    onChange={[Function]}
    readOnly={false}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    htmlFor="radio2"
    role="radio"
  >
    Radio 2
  </label>
</div>
`;

exports[`Radio Button #render() should render radio button with dataAttrs for input 1`] = `
<div
  className="ui-radio test-class"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    data-my-attr-for-input="some-attr-for-input"
    disabled={false}
    id="radio1"
    onChange={[Function]}
    readOnly={false}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    htmlFor="radio1"
    role="radio"
  >
    Radio 1
  </label>
</div>
`;

exports[`Radio Button #render() should render radio button with dataAttrs for label 1`] = `
<div
  className="ui-radio test-class"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    disabled={false}
    id="radio1"
    onChange={[Function]}
    readOnly={false}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    data-my-attr-for-label="some-attr-for-label"
    htmlFor="radio1"
    role="radio"
  >
    Radio 1
  </label>
</div>
`;

exports[`Radio Button #render() should render radio button with dataAttrs for root element 1`] = `
<div
  className="ui-radio test-class"
  data-my-attr="some-attr"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    disabled={false}
    id="radio1"
    onChange={[Function]}
    readOnly={false}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    htmlFor="radio1"
    role="radio"
  >
    Radio 1
  </label>
</div>
`;

exports[`Radio Button #render() should render radio button with provided className 1`] = `
<div
  className="ui-radio test-class"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    disabled={false}
    id="radio1"
    onChange={[Function]}
    readOnly={false}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    htmlFor="radio1"
    role="radio"
  >
    Radio 1
  </label>
</div>
`;
